common/mariadb/templates/pre-change-job.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if or .Values.databases .Values.users }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ include "fullName" . }}-pre-change"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ default "unknown" .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    "helm.sh/hook": pre-upgrade,pre-rollback
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{ include "fullName" . }}-pre-change"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ .Values.name }}-db-exec
+      containers:
+      - name: pre-change-job
+        image: "{{ required ".Values.global.dockerHubMirrorAlternateRegion is missing" .Values.global.dockerHubMirrorAlternateRegion }}/{{ .Values.pre_change_hook.image }}:{{ .Values.pre_change_hook.image_version }}"
+        command:
```

Comment:
Is there any chance we can do that easier?
e.g. by adding k8s lifecycle command with some failsafe sql? https://github.com/sapcc/helm-charts/blob/master/common/mariadb/templates/deployment.yaml#L55-L58

Reply:
I listed all the options in #3595. We probably move the discussions of alternatives there.

Just as a TLDR:
* `postStart` means restart. I want just want to get a user deployed out without a restart.
* While `postStart` running, the `Service` is down, as the container is [not marked running](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-state-running), so it adds to the downtime